src/client/user/components/CreateUrlModal/CreateLinkForm.tsx

```diff
@@ -28,57 +26,34 @@ import { CollapsibleMessageType } from '../../../app/components/CollapsibleMessa
 import FileInputField from '../../widgets/FileInputField'
 import userActions from '../../actions'
 import { GAEvent } from '../../../app/util/ga'
+import { GoGovReduxState } from '../../../app/reducers/types'
+import FormStartAdorment, { TEXT_FIELD_HEIGHT } from './FormStartAdorment'
 
-// Height of the text field in the create link dialog.
-const TEXT_FIELD_HEIGHT = 44
-
-const FormStartAdorment = ({ children }) => {
-  const classes = useCreateLinkFormStyles({
-    textFieldHeight: TEXT_FIELD_HEIGHT,
-  })
-  return (
-    <InputAdornment className={classes.startAdorment} position="start">
-      <Typography className={classes.startAdormentText} color="textSecondary">
-        {children}
-      </Typography>
-    </InputAdornment>
-  )
+type CreateLinkFormProps = {
+  onSubmitLink: () => {},
+  onSubmitFile: (file: File | null) => {},
 }
 
-const mapStateToProps = (state) => ({
-  shortUrl: state.user.shortUrl,
-  longUrl: state.user.longUrl,
-  isUploading: state.user.isUploading,
-  createShortLinkError: state.user.createShortLinkError,
-  uploadFileError: state.user.uploadFileError,
-})
-
-const mapDispatchToProps = (dispatch) => ({
-  setShortUrl: (shortUrl) => dispatch(userActions.setShortUrl(shortUrl)),
-  setLongUrl: (longUrl) => dispatch(userActions.setLongUrl(longUrl)),
-  setRandomShortUrl: () => dispatch(userActions.setRandomShortUrl()),
-  setUploadFileError: (error) =>
-    dispatch(userActions.setUploadFileError(error)),
-  setCreateShortLinkError: (error) =>
-    dispatch(userActions.setCreateShortLinkError(error)),
-})
-
-function CreateLinkForm({
+const CreateLinkForm: FunctionComponent<CreateLinkFormProps> = ({
   onSubmitLink,
-  shortUrl,
-  setShortUrl,
-  longUrl,
-  setLongUrl,
-  setRandomShortUrl,
-  isUploading,
   onSubmitFile,
-  uploadFileError,
-  setUploadFileError,
-  createShortLinkError,
-  setCreateShortLinkError,
-}) {
+}) => {
+  const shortUrl = useSelector((state: GoGovReduxState) => state.user.shortUrl)
+  const longUrl = useSelector((state: GoGovReduxState) => state.user.longUrl)
+  const isUploading = useSelector((state: GoGovReduxState) => state.user.isUploading)
+  const createShortLinkError = useSelector((state: GoGovReduxState) => state.user.createShortLinkError)
+  const uploadFileError = useSelector((state: GoGovReduxState) => state.user.uploadFileError)
```

Comment:
this looks like a good refactor. i have a couple of questions:

1. is it possible for TypeScript to infer the type of `state` automatically without explicitly stating `GoGovReduxState`?
2. would it be cleaner to destructure the argument so that the syntax is like below?

```typescript
useSelector(({user}) => user.shortUrl)
```

Reply:
Unfortunately without stating GoGovReduxState, they will implicitly think that it is defaultrootstate, which is not what we want to reference to. 

So without stating GoGovReduxState, we cannot deconstruct it as well because all our states (user,login,home etc) cannot be found on defaultrootstate

Reply:
ok. there's probably a canonical solution to address this, but we can go with this for now. lmk if you find a better solution.